### ISSUE TITLE
error autorefresh tableau

### ISSUE THREAD

**Issue by kirill3000** [blynkkk/clickhouse4j#46]
java.lang.UnsupportedOperationException

![image](https://user-images.githubusercontent.com/22203338/64711518-62edb400-d4c2-11e9-8bf0-7430cae5938e.png)

**Comment by doom369** [comment]
@kirill3000 Hello. Could you please provide more info. What operation is performed? What tool are you using? Maybe some logs?

**Comment by xadhatter** [comment]
Hey @doom369 I am getting this error as well when using Tableau Desktop v2019.1.12 from a Windows machine.  It only occurs on tables that have a `Date` datatype, `DateTime` works fine.  I get the same error message using the official Yandex JDBC driver as well.

I don't see anything useful in the logs other than the error message provided.  Any help would be greatly appreciated.  The driver is working great otherwise, thanks for all the hard work!

**Comment by doom369** [comment]
@xadhatter thanks for the details. Do you have some log of queries that Tableau performs? I would like to know what exactly query Tableau perform when it fails. It will help a lot.

**Comment by xadhatter** [comment]
@doom369 , here is my test setup:
```sql
CREATE TABLE test (
	id UInt32,
	date Date
)
ENGINE = MergeTree()
ORDER BY (id);

INSERT INTO test VALUES (0, '2020-01-01');
```

And the query that is coming from Tableau:
```sql
SELECT
	TOP 1000 1 AS `Number_of_Records`,
	`test`.`date` AS `date`,
	`test`.`id` AS `id`
FROM
	`default`.`test` `test`
```

If I plug that query into DBeaver using your driver it works fine.  So I assume Tableau is calling some method in the JDBC driver to deal with the `Date` field that isn't implemented yet.  I'll try to get the stack trace for you.

**Comment by doom369** [comment]
@xadhatter date is implemented within driver, so that's why I need precise method call to look into. If you don't know where to look in the Tableau - you can take a look into Clickhouse log:

`/var/log/clickhouse-server/clickhouse-server.log`
or here
`/var/log/clickhouse-server/clickhouse-server.err.log`

Those logs will tell you about precise query that are exectued and thus we will know what causes the issue. I'm eager to help, but need some entry point to start.

**Comment by xadhatter** [comment]
@doom369 Thanks so much for the fast responses.  The query above is the exact query that Tableau is making. No errors on the clickhouse side.

Tableau is a bit of a black box, gotta love debugging closed source software.  Here is the exact log coming out from Tableau...
```json
{
    "ts": "2020-02-10T10:31:41.783",
    "pid": 8968,
    "tid": "2138",
    "sev": "warn",
    "req": "-",
    "sess": "-",
    "site": "-",
    "user": "-",
    "k": "compute-task-error",
    "v": {
        "msg": "java.lang.UnsupportedOperationException\nThere was a Java error.\n",
        "stack": "disabled",
        "task-name": "class BuildDataSourcePresModelTask"
    }
}
```
I'm trying to figure out how to enable "stack" which I _hope_ will give us the Java stacktrace and the exact method, to make your life waay too easy ;) but struggling.  I will reach out to Tableau support shortly if I can't figure it out.

**Comment by xadhatter** [comment]
@doom369 I've opened a support ticket with Tableau.  I'll keep you posted.

**Comment by xadhatter** [comment]
@doom369 I win!  Did a dirty hack:
```java
try {
	throw new UnsupportedOperationException();
} catch (UnsupportedOperationException ex) {
	try (PrintWriter pw = new PrintWriter("C:\\Users\\xadhatter\\Downloads\\jdbcerror.log\"")) {
		ex.printStackTrace(pw);
	} catch (FileNotFoundException e) { }
	throw ex;
}
```
...and got...
```
java.lang.UnsupportedOperationException
	at cc.blynk.clickhouse.response.AbstractResultSet.getDate(AbstractResultSet.java:1864)
	at com.tableausoftware.data.ProtobufResultSet.addValue(ProtobufResultSet.java:268)
	at com.tableausoftware.data.ProtobufResultSet.addRow(ProtobufResultSet.java:202)
	at com.tableausoftware.data.ProtobufResultSet.fetchRows(ProtobufResultSet.java:154)
	at com.tableausoftware.data.ProtobufResultSet.nextBlock(ProtobufResultSet.java:179)
```

**Comment by xadhatter** [comment]
Realized the line numbers will be off.  It's this method, https://github.com/blynkkk/clickhouse4j/blob/master/src/main/java/cc/blynk/clickhouse/response/AbstractResultSet.java#L641

**Comment by doom369** [comment]
@xadhatter pushed the fix and made temporary build:

https://github.com/blynkkk/clickhouse4j/releases/download/clickhouse4j-1.4.3/clickhouse4j-1.4.4-SNAPSHOT.jar

Please let me know if it works.

**Comment by xadhatter** [comment]
@doom369 It works!  Thanks so much for the speedy fix.  I'm new to ClickHouse but really liking it.  Nice to see such a great community forming around it.

**Comment by doom369** [comment]
@xadhatter cool. We just released version 1.4.4. Thanks again for the details.
